### PR TITLE
Add generic Dockerfile and docker-compose.yml for pion examples/turn-server/simple

### LINKED ISSUE #151: Dockerize it
Hi,

I was trying to use it in docker, however I failed on building the go image - probably needed to import some specific go libs. 
I'm a total newbie to GO so I failed.

Would you be able to prepare a dockerfile for the examples?
One generic dockerfile probably would do the trick.

I suppose it would be much more usable if it was easier to deploy :)

### LINKED ISSUE #238: Dockerfile and docker-compose files for the examples.
## Summary

I tried the turn server example for one of the webrtc projects. It came to my observation, there's no docker-compose file for the examples to try out really quickly, so I would like people to have a docker-compose file for using the turn server quickly. I have made a Dockerfile and docker-compose file, please let me know if I can create a PR with the same. 

## Motivation

+ Why are we doing this?
  + quick trial through docker-compose
  + a base Dockerfile, dock-compose file which can be extended according to user's needs.
  + easy to incorporate with any dockerized environment.
+ What use cases does it support? 
  + installation of turn server in any dockerized environment.
+ What is the expected outcome? 
  + Additional docker file and docker-compose files in examples
+ Is this available in other implementations? 
  + I was not able to find a well-maintained version of the docker-compose file.

### PR DESCRIPTION
**Pull request by hthetiot**
#### Description

add generic Dockerfile and docker-compose.yml for pion examples/turn-server/simple

Usage:
> git clone https://github.com/pion/turn.git pion-turn
> cd pion-turn/examples/turn-server/simple
> docker build -t pion-turn -f Dockerfile .
> docker run -e REALM="localhost" -e USERS="username=password" -e UDP_PORT="3478" -e PUBLIC_IP="127.0.0.1" -p 3478:3478 pion-turn

#### Reference issue
Fixes #151
Fixes #238

### CODE REVIEW

**Comment by hthetiot** [comment]
Note:
I dont use `go get` and `go install` because it's too slow due and cause `use of internal package not allowed`
> RUN go get github.com/pions/turn
> RUN go install github.com/pions/turn/cmd/simple-turn

See https://github.com/golang/go/issues/13078#issuecomment-543858784

File: examples/turn-server/simple/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    ports:
+      # STUN
+      - "3478:3478"
+      # TURN
+      - "49152-65535:49152-65535"
+    cap_add:
+      - NET_ADMIN
+      - NET_RAW
+   
+#networks:
```
Comment (stv0g) [inline examples/turn-server/simple/docker-compose.yml:29]: Can we remove the `networks` sections here if they are not used?

File: examples/turn-server/simple/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+version: "3.1"
+
+pion-turn:
+  freeswitch:
+    container_name: pion-turn
+     image: pion-turn:${VERSION:-latest}
+    build:
+      context: ./
+    tty: true
```
Comment (stv0g) [inline examples/turn-server/simple/docker-compose.yml:9]: I am wondering: why is this needed?

File: examples/turn-server/simple/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+      - REALM=${PION_TURN_REALM:-localhost}
+      - USERS=${PION_TURN_USERS:-username=password}
+      - PUBLIC_IP=${PION_TURN_PUBLIC_IP:-127.0.0.1}
+      - UDP_PORT=${PION_TURN_UDP_PORT:-3478}
+    #networks:
+    #  - backend
+    network_mode: host
+    ports:
+      # STUN
+      - "3478:3478"
```
Comment (stv0g) [inline examples/turn-server/simple/docker-compose.yml:22]: Can we use the environment variables here?

File: examples/turn-server/simple/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,50 @@
+FROM golang:1.10
```
Comment (stv0g) [inline examples/turn-server/simple/Dockerfile:1]: How about we use a more recent Go version?
As far as I am aware those old versions are EOL.
Do they are even still receive security updates?

**Review by stv0g — CHANGES_REQUESTED** [review]
Hi @hthetiot,

I added some comments to your code. Can you have a look?

**Comment by stv0g** [comment]
Hey @hthetiot,

are you still interested in getting this merged? I would be happy to support you with this PR.

**Comment by hthetiot** [comment]
> Hey @hthetiot,
> 
> are you still interested in getting this merged? I would be happy to support you with this PR.

I will look into to PR feedback, Sorry missed the notifications.

File: examples/turn-server/simple/Dockerfile (hunk 4 above)
Comment (hthetiot) [reply examples/turn-server/simple/Dockerfile]: used latest instead

File: examples/turn-server/simple/docker-compose.yml (hunk 3 above)
Comment (hthetiot) [reply examples/turn-server/simple/docker-compose.yml]: Done

File: examples/turn-server/simple/docker-compose.yml (hunk 2 above)
Comment (hthetiot) [reply examples/turn-server/simple/docker-compose.yml]: removed, should work without.

File: examples/turn-server/simple/docker-compose.yml (hunk 1 above)
Comment (hthetiot) [reply examples/turn-server/simple/docker-compose.yml]: Done

**Comment by stv0g** [comment]
Perfect :) Can you squash your changes into a single commit and fix the commit message at the same time?

Here are the rules the CI checks for the commit message:

```
The preceding commit message is invalid
it failed 'Separate subject from body with a blank line' of the following checks

* Separate subject from body with a blank line
* Limit the subject line to 50 characters
* Capitalize the subject line
* Do not end the subject line with a period
* Wrap the body at 72 characters
```

**Comment by stv0g** [comment]
Hey @hthetiot,

I squashed your changes. However, in order to merge them you need to grant me push access to your branch.
There should be a little checkbox somewhere for you in this PR here..

**Comment by hthetiot** [comment]
> Hey @hthetiot,
> 
> I squashed your changes. However, in order to merge them you need to grant me push access to your branch. There should be a little checkbox somewhere for you in this PR here..

Thx for squash, I found that on Mac docker `compose up` fail due port range (49152-65535) taken on TURN  and generally this range of port cause issue in many environments.

Can you hold on PR so I add a short Readme and add custom port range for TURN in env.

If you can point me to a custom TURN port range example with pion that may help.

**Comment by stv0g** [comment]
Hi @hthetiot 

Sure :)

Here is an example TURN server which which binds to a user defined port range: https://github.com/pion/turn/blob/master/examples/turn-server/port-range/main.go

**Comment by hthetiot** [comment]
> Hi @hthetiot
> 
> Sure :)
> 
> Here is an example TURN server which which binds to a user defined port range: https://github.com/pion/turn/blob/master/examples/turn-server/port-range/main.go

Perfect, I will update example asap.

**Comment by stv0g** [comment]
Hi @hthetiot,

is there something I can support you with to get this merged?

**Comment by hthetiot** [comment]
> Hi @hthetiot,
> 
> is there something I can support you with to get this merged?

Well. I wanted to add dynamic TURN port, I just need more time to do so, if you want I can give you acess to the branch.

File: examples/turn-server/simple/docker-compose.yml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+    environment:
+      - VERSION=${PION_TURN_VERSION:-master}
+      - REALM=${PION_TURN_REALM:-localhost}
+      - USERS=${PION_TURN_USERS:-username=password}
+      - PUBLIC_IP=${PION_TURN_PUBLIC_IP:-127.0.0.1}
+      - UDP_PORT=${PION_TURN_UDP_PORT:-3478}
+    network_mode: host
+    ports:
+      # STUN
+      - "${PION_TURN_UDP_PORT:-3478}:${PION_TURN_UDP_PORT:-3478}"
```
Comment (hthetiot) [inline examples/turn-server/simple/docker-compose.yml:19]: TCP?

Comment (stv0g) [reply examples/turn-server/simple/docker-compose.yml:19]: Yes, could be also added here.

Comment (hthetiot) [reply examples/turn-server/simple/docker-compose.yml:19]: was merged without TCP and custom TURN port.
I did not have bandwith to work on it.
I may to another PR later for thoses.

Comment (stv0g) [reply examples/turn-server/simple/docker-compose.yml:19]: No problem. We are not in a rush. I just wanted to have this merged so we can improve it later if somebody finds time to improve this.
